static/js/src/main.js

```diff
@@ -12,15 +12,30 @@ var detectionOptions = {
     htmlTag: document.documentElement
 };
 
-i18n.use(XHR)
-    .use(lngDetector)
-    .init({
-        nsSeparator: false,
-        keySeparator: false,
-        interpolation: {
-            prefix: "__",
-            suffix: "__"
-        },
-        backend: backendOptions,
-        detection: detectionOptions
+function init_i18n(callback) {
+    i18n.use(XHR)
+        .use(lngDetector)
+        .init({
+            nsSeparator: false,
+            keySeparator: false,
+            interpolation: {
+                prefix: "__",
+                suffix: "__"
+            },
+            backend: backendOptions,
+            detection: detectionOptions
+    }, callback);
+}
+
+i18n.initialized = false;
+init_i18n(function () {
+    i18n.initialized = true;
 });
+
+i18n.ensure_i18n = function (callback) {
+    if (i18n.initialized) {
+        callback();
+    } else {
+        init_i18n(callback);
```

Comment:
But now I guess I have the question of why we're calling `init_i18n` multiple times -- it seems like the additional calls will result in the i18n initialization code being called multiple times, which is potentially inefficient.

I think we might want to instead just have:
```
callbacks = [];
init_i18n(function () {
     i18n.initialized = true;
    // add code to loop through callbacks calling them
});

i18n.ensure_i18n = function (callback) {
     if (i18n.initialized) {
         callback();
     } else {
         callbacks.push(callback);
    }
}￼
```